src/Illuminate/Support/Collection.php

```
@@ -955,6 +955,10 @@ public function split($numberOfGroups)
      */
     public function chunk($size)
     {
+        if ($size == 0) {
```

We should throw an invalid argument exception for less than 0 also.